bftengine/src/communication/PlainUDPCommunication.cpp

```diff
@@ -337,8 +347,7 @@ class PlainUDPCommunication::PlainUdpImpl {
 
   void
   recvThreadRoutine() {
-    bool sRunning = true;       /** Indicates whether we're still running */
-
+    running = true;
```

Comment:
This is set in `Start`, after the thread is created. I think that setting it here actually introduces a race condition:

1. Start starts this thread.
2. Start sets running = true.
3. Stop is called.
4. Stop sets running = false.
5. This thread finally starts executing, and sets running = true.
6. Stop is left waiting to join this thread, but nothing will set running to false.

Who knows why someone would call Start and then immediately call Stop in a way that would produce the above scheduling. But, I think it can be fixed anyway, by removing this `running = true` within the thread, and moving the `running = true` in `Start` to before the thread creation (the call to `startRecvThread`).